src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -53,34 +56,58 @@ public Dissector(String mapping) {
         Collections.sort(saveableFields, new FieldComparator());
     }
 
-    public int dissect(byte[] source, Map<String, Object> map) {
-        int pos = dissectValues(source);
-        ValueResolver resolver = new ValueResolver(source, fieldValueRefMap);
+    public int dissect(byte[] source, Map<String, Object> keyValueMap) {
+        ResolveValue resolveValue = resolve(source);
```

Comment:
I looked once again at `#dissect()` and maybe you're better off inlining this method and getting rid of `ResolveValue` alltogether. It's really just a holder for your return value.

> What is the norm for returning multiple values from a method?

I totally understand where are you coming from. In the Ruby (and also in the Python) world this is a normal thing but in Java it's quite unusual and if you ask this question it usually indicates that you have a problem in your design.

You usually don't return multiple (unrelated) values from a method. Instead you return an object which should "logically group" its properties. Here is an example to clarify my point:

* Good: `Person` with `firstName` and `lastName`
* Bad: `DataHolder` with `purchaseOrder` and `Date`

In the latter case, people sometimes create their own `Pair` or `Tuple` class and use that but usually that's already a sign that something is wrong.